### ISSUE TITLE
Rename the header?

### ISSUE THREAD

**Issue by icculus** [libsdl-org/SDL_gesture#7]
So right now, you would `#include "SDL_gesture.h"`, and on SDL3, where there is no such file, it will choose the standalone header from this project.

If you are using SDL2, and everything goes right:
- You would include SDL.h first (as required by SDL_gesture.h).
- SDL.h would include everything else it wants.
- You would include SDL_gesture.h, and the compiler would choose the standalone header.
- The standalone header notices that you are using SDL2 headers, define a few things, and let SDL2 do all the work.

But there's two ways this can go wrong:
- You include SDL.h
- SDL.h includes SDL_gesture.h, but gets the standalone header instead of SDL2's.
- Things break.

Or:
- You include SDL.h
- You include SDL_gesture.h
- You end up with SDL2's SDL_gesture.h and Gesture_* symbols do not get defined.

The easiest fix here is to not call this thing SDL_gesture.h. Maybe SDL_gesture_standalone.h? If we do this, I need a better name.

(If we don't do this, someone explain to me how those two failing cases are totally ridiculous and wouldn't happen in real life, please.)

**Comment by madebr** [comment]
If the `-I` of this project happens before the `-I` of SDL2, then everything is fine.
If the `-I` of SDL2 is before the one from this project, this one gets blocked.
This public dependency on order of includes is undesired.

<details>

<summary>Order of -I matters</summary>

```sh
mkdir -p inc/proj

cat >inc/proj/proj.h <<EOF
#ifndef proj_h
#define proj_h
int proj_h_start(void);
#include "proj_something.h"
int proj_h_end(void);
#endif
EOF

cat >inc/proj/proj_something.h <<EOF
#ifndef proj_something_h
#define proj_something_h
int proj_something_h_start(void);
int proj_something_h_end(void);
#endif
EOF

cat >inc/proj_something.h <<EOF
#ifndef proj_something_clone_h
#define proj_something_clone_h
int proj_clone_something_h_start(void);
int proj_clone_something_h_end(void);
#endif
EOF

cat >user.c <<EOF
#include "proj.h"
#include "proj_something.h"
EOF

set -x
: "This is okay:"
cpp -Iinc -Iinc/proj -P user.c

: "This breaks:"
cpp -Iinc/proj -Iinc -P user.c
```

</details>

What about `SDL_gesture_compat.h`?

**Comment by icculus** [comment]
Yeah, I like SDL_gesture_compat. Waiting on a greenlight from @slouken and then I'll rename the header _and_ repo to that.

**Comment by slouken** [comment]
What if this header goes to SDL3/SDL_gesture.h and we leave the SDL_* naming unchanged? If you're using SDL2, you just use the existing header, and if you're using SDL3, you drop this in?

BTW, we've spent a lot of cycles on a thing that we have no idea if people will use. I'm kind of inclined to wait and see if people ask for it, before going to all this work.

**Comment by icculus** [comment]
We can leave it and wait for feedback, but I definitely wouldn't put it back in SDL3.
